docs/plugins/pluggable_rule_engine.md

```diff
@@ -32,31 +32,30 @@ Because the rule engine plugins are just one of many iRODS plugin types, the REP
+However, any that *do* match a particular PEP will be executed in the order in which they are loaded.  If a rule is matched, then it will fire and its return code will be interpreted by the REPF.  If it fails, then the operation fails as well and an error is returned to the client (or to the log when run by the [delay execution server](#delay-execution).)
 
-If there is more than one matching rule for a particular PEP, the first one loaded will fire first.  If it succeeds, then the others are ignored.  If the first one fails, then the next matching rule is fired.  If it fails, then the framework will continue to "fall through" until there are no more matching rules.  The return code of the last matching rule will be the one that is returned.
+The REPF can be instructed to do different things based on the code returned from a rule. See the following table:
 
-![Rule Engine Plugin Framework Diagram](../images/rule_engine_plugin_framework_diagram.jpg)
+| Symbolic Code Name         | Code      | Behavior |
+| -------------------------- | --------- | -------- |
+| RULE_ENGINE_CONTINUE       | 5000000   | Allow the next **rule engine plugin** to process the PEP.  This does not apply to rules within the same rule engine plugin. |
+| RULE_ENGINE_SKIP_OPERATION | 5001000   | Do not execute the API operation following the **pre** PEP.  This code is only honored by **pre** PEPs.  If returned from a PEP that isn't a **pre** PEP, the server will log a warning and ignore the instruction. |
+| SYS_NOT_SUPPORTED          | -66000    | Synonymous with RULE_ENGINE_CONTINUE. |
 
-This "fall through" mechanism also applies across rule engine plugins (meaning, across language as well).
+`RULE_ENGINE_CONTINUE` is important for implementing maintainable policy.  It allows rule code to be split across rule engine plugins. Doing this allows policy implementers to separate concerns and focus on smaller chunks of code.  Basic usage is demonstrated in the example below.
 
-There are three scopes where the "fall through" will occur:
-
-- within the framework itself, falling from one plugin to another
-- within a plugin, falling from one rulebase to another
-- within a rulebase, falling from one rule to another
-
-This nested scope defines the order in which identically named rules take precedent on a particular iRODS server.
+`RULE_ENGINE_SKIP_OPERATION` allows policy implementers to skip or replace the behavior of an API operation.  Care must be taken when using this instruction.  The Consortium highly recommends avoiding it unless absolutely necessary.
 
 ### Example
 
-Consider the following `server_config.json` with the accompanying `example.re`, `core.re` and `core.py` rulebases:
+Consider the following `server_config.json` with the accompanying `core.re` and `core.py`.
```

Comment:
switch the order to be... `core.py` and `core.re`

as that is how they are displayed and explained below...

Reply:
Done.